include/pybind11/pytypes.h

```diff
@@ -1012,6 +1014,12 @@ inline str::str(const bytes& b) {
     m_ptr = obj.release().ptr();
 }
 
+#if PY_MAJOR_VERSION < 3
+using native_str = bytes;
+#else
+using native_str = str;
+#endif
+
```

Comment:
Ah, very good, thanks for pointing that out! I would never have guessed from the name what it does, but now I know :-)

Reply:
If you/we want to provide this as API to pybind11 users, we should provide a typedef, though, rather than have them use this macro.
If so, I'm just not entirely convinced about `native_str` as a name (though I don't now, immediately have a better suggestion).